web/src/components/Storyblok/StoryblokSectionNav/utils/Utils.tsx

```diff
@@ -0,0 +1,97 @@
+type Link = {
+	childLinks?: Link[];
+	id: string;
+	slug: string;
+	parent_id: string;
+	is_folder: boolean;
+	is_startpage: boolean;
+};
+
+
+export const fetchLinksFromStoryblok = async (
+	queryParams: Record<string, string>
+): Promise<Link[]> => {
+	const queryString = new URLSearchParams(queryParams).toString();
+	const res = await fetch(
+		`https://api.storyblok.com/v2/cdn/links?${queryString}`
+	);
+	const data = await res.json();
+	const dataArray: Link[] = Object.values(data.links);
+	return dataArray;
+};
+export const fetchParentAndSiblingLinks = async (
+	token: string,
+	parentID: string,
+	slug: string
+) => {
+	// Fetch sibling links first
+	const siblingsLinksArray = await fetchLinksFromStoryblok({
+		version: "published",
+		token,
+		with_parent: parentID || "",
+		per_page: "1000",
+	});
+	const parentAndSiblingLinksArray = await reUseFetchingLogic(
+		token,
+		slug,
+		siblingsLinksArray
+	);
+	return { siblingsLinksArray, parentAndSiblingLinksArray };
+};
+export const filterFunctionForTreeStructure = (
```

Comment:
I'd done something similar when I'd had a look at refactoring but gone about it slightly differently - worth having a nosy for interest but no strong feelings as to which way we go with! https://github.com/nice-digital/next-web/commit/797c2c1f2405ddeb2b60133a7b59acaa76d313e0